plugins/modules/keycloak_realm_key.py

```diff
@@ -120,7 +123,7 @@
                       key must match O(config.algorithm) and O(provider_id).
                     - If you want Keycloak to automatically generate a certificate using your private key
                       then set this to an empty string.
-                required: true
+                default: ''
```

Comment:
I can't remember the details of why this is/was set to _required: true_. Maybe the idea was to be explicit about "I don't want to pass a certificate of my own" instead of implicitly assuming it. If the integration tests do not fail then I'm not opposed to this change, though.

Reply:
I checked out this API call today and I can confirm that if the cert is an empty string then keycloak will generate a new one. If the cert is omitted entirely from the json object then keycloak will leave the cert alone. @felixfontein it would seem like we should address this in another PR because neither this PR or the existing functionality seems correct here. I can remove the default to match the existing behaviour. I would imagine that most people would probably like to manage their own cert along with the key though...

Reply:
There is no "correct", there is only no "all possible use-cases are supported". I would either try to support all possible use-cases (which includes "keep the certificate as-is"), or not change it (which the PR in its latest form does).

I agree, supporting the "keep the certificate as-is" use-case should better be done in a separate PR.

Reply:
Excellent assessment. Let's get this merged and move on then.